Makefile

```diff
@@ -0,0 +1,49 @@
+.PHONY: install-poetry
+## Bootstrap installing poetry
+install-poetry:
+	curl -sSL https://install.python-poetry.org | python3 -
+
+.PHONY: install
+## Install project and dependencies
+install:
+	@poetry install
+
+.PHONY: format
+## Format the code using black and isort
+format:
+	@poetry run black src tests
+	@poetry run isort .
+
+.PHONY: lint
+## Lint the code using mypy and flake8
+lint:
+	@poetry run mypy src
+	@poetry run flake8 src
+
+.PHONY: gen-test
+## Generate synchronous test scripts from async test scripts. 
+gen-test:
+	@bash tests/scripts/sync_from_async.sh
+
+.PHONY: test
+## Run unit and integration tests with pytest
+test:
+	@poetry run pytest
+
+.PHONY: precommit
+## Run format, lint, and test as a step before committing.
+precommit: gen-test format lint test
+	@echo.
+
+.PHONY: clean
+## Remove intermediate files
+clean:
+	@rm -rf dist .mypy_cache .pytest_cache
+	@find -name "*__pycache__*" | xargs rm -rf
+
+
+.DEFAULT_GOAL := show-help
```

Comment:
Nice good idea. I usually have build be default but help makes more sense I like forced you to be explicit

Reply:
I think the `precommit` target is close to what `build` means. I like your idea though and added an `all` target which does the format-lint-test cycle [here](https://github.com/momentohq/client-sdk-python/pull/138/commits/237febe65a5a985358ef218620475b566b4deb7c)

Reply:
Updated contributed documentation in https://github.com/momentohq/client-sdk-python/pull/138/commits/e9f0d5cbc75525601ca5ca992cc46a226c6814c9